core/Tracker/VisitorRecognizer.php

```diff
@@ -231,7 +231,7 @@ private function getVisitFieldsPersist()
             $dimensions = VisitDimension::getAllDimensions();
 
             foreach ($dimensions as $dimension) {
-                if ($dimension->hasImplementedEvent('onExistingVisit')) {
+                if ($dimension->hasImplementedEvent('onExistingVisit') || $dimension->hasImplementedEvent('onAnyGoalConversion')) {
```

Comment:
Not sure it this is the correct solution.
Explanation why I've changed that:

I had following code in a visit dimension
```
    public function onAnyGoalConversion(Request $request, Visitor $visitor, $action)
    {
        return $visitor->getVisitorColumn($this->columnName);
    }
```
That didn't work properly, as the column didn't implement the `onExistingVisit` event (which simply isn't needed) and thus the property wasn't set for an existing visit.

Instead of this change here, I could also add an empty `onExistingVisit` event to the columns. But I think it is more useful this way.

@mattab @tsteur  what do you think?